### PR TITLE
Add system test for 'Sink.create' using a Pubsub topic.

### PR DESCRIPTION
**Pull request by tseaver**
Currently broken because 'Topic.set_iam_policy' returns 503s.  See: #1654.

### CODE REVIEW

File: system_tests/logging_.py [hunk 1]
```diff
@@ -50,8 +51,12 @@ def setUp(self):
         self.to_delete = []
 
     def tearDown(self):
+        from gcloud.exceptions import NotFound
         for doomed in self.to_delete:
-            doomed.delete()
+            try:
+                doomed.delete()
+            except NotFound as error:
+                print('Unable to delete resource: %s' % error)
```
Comment (dhermes) [inline system_tests/logging_.py]: This is my only hang-up and also the reason for the don't-merge label so we are on the same page.

Comment (tseaver) [inline system_tests/logging_.py]: This is due to #1657 (the loggers "created" via the `/entries:write` API cannot be deleted).

**Comment by tseaver** [comment]
Rebased after merging #1658 to the `logging-api` branch.

**Comment by tseaver** [comment]
@jgeewax this issue blocks merging the `logging-api` branch by EOQ (we have consistent system test failures).

**Comment by tseaver** [comment]
@dhermes 1248b66 backs out the workaround for #1657.  PTAL

File: system_tests/logging_.py [hunk 2]
```diff
@@ -51,6 +52,7 @@ def setUp(self):
         self.to_delete = []
 
     def tearDown(self):
+        from gcloud.exceptions import NotFound
```
Comment (dhermes) [inline system_tests/logging_.py]: This import should go?

File: system_tests/logging_.py [hunk 3]
```diff
@@ -175,3 +177,25 @@ def test_create_sink_bigquery_dataset(self):
         sink.create()
         self.to_delete.append(sink)
         self.assertTrue(sink.exists())
+
+    def test_create_sink_pubsub_topic(self):
+        from gcloud import pubsub
+        TOPIC_URI = 'pubsub.googleapis.com/projects/%s/topic/%s' % (
+            Config.CLIENT.project, TOPIC_NAME)
```
Comment (dhermes) [inline system_tests/logging_.py]: Could we construct this from the `Topic` object?

```python
>>> pubsub_client.connection.api_base_url + topic.path
u'https://pubsub.googleapis.com/projects/P/topics/T'
```

Comment (tseaver) [inline system_tests/logging_.py]: ATM, we have punted on #1591, due to your legitimate concerns about inter-package coupling.

Comment (tseaver) [inline system_tests/logging_.py]: I guess we could do that for the system test only.

Comment (dhermes) [inline system_tests/logging_.py]: Yeah that's what I meant (sorry for not being clearer)

Comment (tseaver) [inline system_tests/logging_.py]: @dhermes the destination string cannot have the `https://` prefix. :(

Comment (dhermes) [inline system_tests/logging_.py]: >>> (pubsub_client.connection.api_base_url + topic.path)[len(https://'):]
u'pubsub.googleapis.com/projects/P/topics/T'

Comment (tseaver) [inline system_tests/logging_.py]: That is *definitely* not clearer than the current code.  Not only that, but it relies on the accident that the `netloc` of the `api_base_url` happens to be the string specified in the `projects.sinks.create` API.

**Comment by dhermes** [comment]
Bad rebase?

**Comment by dhermes** [comment]
Continuing our discussion. Yeah it is too clever but a few more lines will get you where you need without having a hardcoded URL in two different places.

```python
url = pubsub_client.connection.api_base_url + topic.path
if url.startswith('https://'):
    url = url[8:]
```

**Comment by tseaver** [comment]
Yup, I hadn't pushed the underlying `logging-api` merge yet (needed to make `topic.setIamPolcy` not raise).

**Comment by dhermes** [comment]
Ping me when this PR looks how you expected it to?

**Comment by tseaver** [comment]
Using `pubsub_client.connection.api_base_url` to compute the prefix of the `destination` string is wrong:  the prefix as defined in the `projects.sink.create` API just happens to match.  I've updated to use `topic.full_name` to compute the suffix, however.

**Comment by tseaver** [comment]
*Sigh*. I'll open a new PR.

**Pull request closed without merging**